ocaml/xapi/certificates_sync.ml

```diff
@@ -0,0 +1,144 @@
+module D = Debug.Make (struct let name = "certificates_sync" end)
+
+open D
+module Unixext = Xapi_stdext_unix.Unixext
+module Date = Xapi_stdext_date.Date
+module G = Gencertlib
+module GP = Gencertlib.Pem
+module GV = Gencertlib.Validation
+open Api_errors
+open Rresult
+
+let ( let* ) = ( >>= )
+
+let is_tls_verification_enabled () =
+  Localdb.get_with_default Constants.tls_verification_enabled "false"
+  |> bool_of_string
+
+let is_self_signed cert =
+  R.return
+    X509.(
+      Distinguished_name.equal (Certificate.subject cert)
```

Comment:
This doesn't necessarily mean that it was actually signed with its own key, or that the signature is valid.
See https://github.com/mirleft/ocaml-x509/blob/master/lib/validation.ml#L322-L330 for a more comprehensive check.
Perhaps you could use  `is_self_signed + X509.Validation.valid_ca`

Reply:
I'm pretty sure that `X509.Validation.valid_ca` will fail for the certificates that are generated with the xapi-generated certificates in all existing versions. Changing this would mean updating how the certs are generated and gating the old certificates.